### PR TITLE
feat(installation, troubleshooting): update network settings

### PR DESCRIPTION
**Pull request by yukkysaito**
## Description
The documentation around the network was expanded because many people get stuck due to insufficient documentation.
I also corrected duplicate descriptions and wrong links.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The Reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: docs/installation/additional-settings-for-developers/index.md [hunk 1]
```diff
@@ -22,16 +22,46 @@ export RCUTILS_CONSOLE_OUTPUT_FORMAT="[{severity} {time}] [{name}]: {message} ({
...
 ```
 
-### Setting up `ROS_DOMAIN_ID`
+If exported `ROS_LOCALHOST_ONLY=1`, `MULTICAST` must be enabled at the loopback address. Use the following command to verify that `MULTICAST` is included.
+
+```bash
+ip link show lo
+# Result sample
+# 1: lo: <LOOPBACK,MULTICAST,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000
+```
```
Comment (kenji-miyake) [inline docs/installation/additional-settings-for-developers/index.md:56]: To follow the standard.

``````suggestion
```console
$ ip link show lo
1: lo: <LOOPBACK,MULTICAST,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000
```
``````

File: docs/installation/additional-settings-for-developers/index.md [hunk 2]
```diff
@@ -22,16 +22,46 @@ export RCUTILS_CONSOLE_OUTPUT_FORMAT="[{severity} {time}] [{name}]: {message} ({
...
+
+If `MULTICAST` is not included, use the following command to enable it.
```
Comment (kenji-miyake) [inline docs/installation/additional-settings-for-developers/index.md:58]: To clarify it a bit more, how about this?

```suggestion
If the word `MULTICAST` is not printed, use the following command to enable it.
```

File: docs/installation/additional-settings-for-developers/index.md [hunk 3]
```diff
@@ -44,13 +74,21 @@ To avoid it, set a different domain ID for each group in your `.bashrc`:
...
 
-For more information, see [here](https://docs.ros.org/en/foxy/Concepts/About-Domain-ID.html#the-ros-domain-id).
+Also confirm that `ROS_LOCALHOST_ONLY` is `0` by using the following command.
 
-### Tuning DDS
+```bash
+echo $ROS_LOCALHOST_ONLY # If output is 1, localhost has priority.
+```
+
+For more information, see [here](https://docs.ros.org/en/humble/Tutorials/Beginner-CLI-Tools/Configuring-ROS2-Environment.html#the-ros-domain-id-variable).
```
Comment (kenji-miyake) [inline docs/installation/additional-settings-for-developers/index.md:83]: https://developers.google.com/style/link-text



```suggestion
For more information, see the [ROS 2 Documentation](https://docs.ros.org/en/humble/Tutorials/Beginner-CLI-Tools/Configuring-ROS2-Environment.html#the-ros-domain-id-variable).
```

File: docs/installation/additional-settings-for-developers/index.md [hunk 4]
```diff
@@ -44,13 +74,21 @@ To avoid it, set a different domain ID for each group in your `.bashrc`:
 export ROS_DOMAIN_ID=X
 ```
 
-For more information, see [here](https://docs.ros.org/en/foxy/Concepts/About-Domain-ID.html#the-ros-domain-id).
+Also confirm that `ROS_LOCALHOST_ONLY` is `0` by using the following command.
 
-### Tuning DDS
+```bash
+echo $ROS_LOCALHOST_ONLY # If output is 1, localhost has priority.
```
Comment (kenji-miyake) [inline docs/installation/additional-settings-for-developers/index.md:80]:
```suggestion
echo $ROS_LOCALHOST_ONLY # If the output is 1, localhost has priority.
```

File: docs/installation/additional-settings-for-developers/index.md [hunk 5]
```diff
@@ -44,13 +74,21 @@ To avoid it, set a different domain ID for each group in your `.bashrc`:
...
+For more information, see [here](https://docs.ros.org/en/humble/Tutorials/Beginner-CLI-Tools/Configuring-ROS2-Environment.html#the-ros-domain-id-variable).
+
+### DDS settings
 
 Autoware uses DDS for inter-node communication. [ROS 2 documentation](https://docs.ros.org/en/humble/How-To-Guides/DDS-tuning.html) recommends users to tune DDS to utilize its capability. Especially, receive buffer size is the critical parameter for Autoware. If the parameter is not large enough, Autoware will failed in receiving large data like point cloud or image.
 
-For example, to execute Autoware with CycloneDDS, prepare config file. A sample config file, named as `cyclonedds_config.xml`, is given below.
+#### Tuning DDS
+
+Unless customized, CycloneDDS is adopted by default. For example, to execute Autoware with CycloneDDS, prepare config file. A sample config file, named as `cyclonedds_config.xml`, is given below.
```
Comment (kenji-miyake) [inline docs/installation/additional-settings-for-developers/index.md:91]:
```suggestion
Unless customized, CycloneDDS is adopted by default. For example, to execute Autoware with CycloneDDS, prepare a config file. A sample config file is given below. Save it as `cyclonedds_config.xml`.
```

File: docs/installation/additional-settings-for-developers/index.md [hunk 6]
```diff
@@ -73,7 +111,7 @@ sudo sysctl -w net.core.rmem_max=2147483647
 
 Refer to [ROS 2 documentation](https://docs.ros.org/en/humble/How-To-Guides/DDS-tuning.html) for more information. Reading user guide for chosen DDS is helpful for more understanding.
```
Comment (kenji-miyake) [inline docs/installation/additional-settings-for-developers/index.md:112]: Although it's not edited in this PR, but to be consistent styles.

```suggestion
For more information, Refer to [ROS 2 documentation](https://docs.ros.org/en/humble/How-To-Guides/DDS-tuning.html). Reading user guide for chosen DDS is helpful for more understanding.
```

File: docs/support/troubleshooting.md [hunk 7]
```diff
@@ -172,6 +171,11 @@ net.core.rmem_default=8388608 // only add if CycloneDDS is confgured
     echo $RMW_IMPLEMENTATION  // if Cyclone DDS is configured, this command will return "rmw_cyclonedds_cpp"
     ```
 
+If that does not work or you need more information, please read followings.
```
Comment (kenji-miyake) [inline docs/support/troubleshooting.md:174]:
```suggestion
If that does not work or you need more information, read the following documents.
```

File: docs/support/troubleshooting.md [hunk 8]
```diff
@@ -152,16 +152,15 @@ When running the Planning Simulator, the most common reason for the map not bein
...
-net.ipv4.ipfrag_time=3  // generic DDS setting
-net.ipv4.ipfrag_high_thresh=134217728 // generic DDS setting
-net.core.rmem_max=2147483647 // only add if CycloneDDS is configured
-net.core.rmem_default=8388608 // only add if CycloneDDS is confgured
+export ROS_LOCALHOST_ONLY=1
+if [ ! -e /tmp/cycloneDDS_configured ]; then
+  sudo sysctl -w net.core.rmem_max=2147483647
+  sudo ip link set lo multicast on
+  touch /tmp/cycloneDDS_configured
+fi
```
Comment (kenji-miyake) [inline docs/support/troubleshooting.md:163]: Memo: Added by https://github.com/autowarefoundation/autoware-documentation/pull/107.

For this part, I want a review from @takam5f2.

**Review by kenji-miyake — COMMENTED** [review]
Thank you so much! :smile: 
Added some comments. Most of them are about styling based on https://autowarefoundation.github.io/autoware-documentation/main/contributing/documentation-guidelines/.

File: docs/installation/additional-settings-for-developers/index.md [hunk 9]
```diff
@@ -22,16 +22,46 @@ export RCUTILS_CONSOLE_OUTPUT_FORMAT="[{severity} {time}] [{name}]: {message} ({
 
 For more options, see [here](https://docs.ros.org/en/rolling/Tutorials/Logging-and-logger-configuration.html#console-output-formatting).
 
-### Enabling localhost-only communication
+## Network settings for ROS 2
+
+This section describes the network settings.
```
[the remark this answers is not in the document]
Comment (yukkysaito) [reply docs/installation/additional-settings-for-developers/index.md]: Fixed at https://github.com/autowarefoundation/autoware-documentation/pull/306/commits/cebfd3c8f86dd3b92ebc9f1607a148f972c35a2b :+1:

File: docs/installation/additional-settings-for-developers/index.md [hunk 10]
```diff
@@ -22,16 +22,46 @@ export RCUTILS_CONSOLE_OUTPUT_FORMAT="[{severity} {time}] [{name}]: {message} ({
...
+ROS 2 employs DDS, and the configuration of ROS2 and DDS will be described separately.
+Please refer to [the official documentation](http://design.ros2.org/articles/ros_on_dds.html) for ROS2 networking concepts.
+
+### ROS 2 network setting
+
+ROS2 multicasts data on the local network by default. When developing in a company, etc., data flows over the local network and there is a possibility of collision.
```
[the remark this answers is not in the document]
Comment (yukkysaito) [reply docs/installation/additional-settings-for-developers/index.md]: Fixed at https://github.com/autowarefoundation/autoware-documentation/pull/306/commits/cebfd3c8f86dd3b92ebc9f1607a148f972c35a2b :+1:

File: docs/installation/additional-settings-for-developers/index.md [hunk 11]
```diff
@@ -22,16 +22,46 @@ export RCUTILS_CONSOLE_OUTPUT_FORMAT="[{severity} {time}] [{name}]: {message} ({
 
 For more options, see [here](https://docs.ros.org/en/rolling/Tutorials/Logging-and-logger-configuration.html#console-output-formatting).
 
-### Enabling localhost-only communication
+## Network settings for ROS 2
+
+This section describes the network settings.
+ROS 2 employs DDS, and the configuration of ROS2 and DDS will be described separately.
```
[the remark this answers is not in the document]
Comment (yukkysaito) [reply docs/installation/additional-settings-for-developers/index.md]: Fixed at https://github.com/autowarefoundation/autoware-documentation/pull/306/commits/cebfd3c8f86dd3b92ebc9f1607a148f972c35a2b :+1:

File: docs/installation/additional-settings-for-developers/index.md [hunk 12]
```diff
@@ -22,16 +22,46 @@ export RCUTILS_CONSOLE_OUTPUT_FORMAT="[{severity} {time}] [{name}]: {message} ({
...
+Please refer to [the official documentation](http://design.ros2.org/articles/ros_on_dds.html) for ROS2 networking concepts.
```
[the remark this answers is not in the document]
Comment (yukkysaito) [reply docs/installation/additional-settings-for-developers/index.md]: Fixed at https://github.com/autowarefoundation/autoware-documentation/pull/306/commits/cebfd3c8f86dd3b92ebc9f1607a148f972c35a2b :+1:

File: docs/installation/additional-settings-for-developers/index.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (yukkysaito) [reply docs/installation/additional-settings-for-developers/index.md]: Fixed at https://github.com/autowarefoundation/autoware-documentation/pull/306/commits/cebfd3c8f86dd3b92ebc9f1607a148f972c35a2b :+1:

File: docs/installation/additional-settings-for-developers/index.md [hunk 13]
```diff
@@ -22,16 +22,46 @@ export RCUTILS_CONSOLE_OUTPUT_FORMAT="[{severity} {time}] [{name}]: {message} ({
...
+ROS2 multicasts data on the local network by default. When developing in a company, etc., data flows over the local network and there is a possibility of collision.
+
+- Localhost-only communication
+- Same domain only communication on the local network
+
+Unless you plan to use multiple host computers on the local network, localhost-only communication is recommended.
+
+#### Enabling localhost-only communication
 
 By default, ROS 2 communicates using multi-cast, which may unnecessarily increase the network traffic.
```
[the remark this answers is not in the document]
Comment (yukkysaito) [reply docs/installation/additional-settings-for-developers/index.md]: Fixed at https://github.com/autowarefoundation/autoware-documentation/pull/306/commits/cebfd3c8f86dd3b92ebc9f1607a148f972c35a2b :+1:

File: docs/installation/additional-settings-for-developers/index.md [hunk 14]
```diff
@@ -22,16 +22,46 @@ export RCUTILS_CONSOLE_OUTPUT_FORMAT="[{severity} {time}] [{name}]: {message} ({
...
 To avoid it, write the following in your `.bashrc`:
```
[the remark this answers is not in the document]
Comment (yukkysaito) [reply docs/installation/additional-settings-for-developers/index.md]: Fixed at https://github.com/autowarefoundation/autoware-documentation/pull/306/commits/cebfd3c8f86dd3b92ebc9f1607a148f972c35a2b :+1:

File: docs/installation/additional-settings-for-developers/index.md [hunk 15]
```diff
@@ -22,16 +22,46 @@ export RCUTILS_CONSOLE_OUTPUT_FORMAT="[{severity} {time}] [{name}]: {message} ({
...
+For more information, see [here](https://docs.ros.org/en/humble/Tutorials/Beginner-CLI-Tools/Configuring-ROS2-Environment.html#the-ros-localhost-only-variable).
```
[the remark this answers is not in the document]
Comment (yukkysaito) [reply docs/installation/additional-settings-for-developers/index.md]: Fixed at https://github.com/autowarefoundation/autoware-documentation/pull/306/commits/cebfd3c8f86dd3b92ebc9f1607a148f972c35a2b :+1:

File: docs/installation/additional-settings-for-developers/index.md [hunk 16]
```diff
@@ -22,16 +22,46 @@ export RCUTILS_CONSOLE_OUTPUT_FORMAT="[{severity} {time}] [{name}]: {message} ({
...
 
 ```bash
 export ROS_LOCALHOST_ONLY=1
 ```
 
-### Setting up `ROS_DOMAIN_ID`
+If exported `ROS_LOCALHOST_ONLY=1`, `MULTICAST` must be enabled at the loopback address. Use the following command to verify that `MULTICAST` is included.
```
[the remark this answers is not in the document]
Comment (yukkysaito) [reply docs/installation/additional-settings-for-developers/index.md]: Fixed at https://github.com/autowarefoundation/autoware-documentation/pull/306/commits/cebfd3c8f86dd3b92ebc9f1607a148f972c35a2b :+1:

File: docs/installation/additional-settings-for-developers/index.md (hunk 1 above)
Comment (yukkysaito) [reply docs/installation/additional-settings-for-developers/index.md]: Fixed at https://github.com/autowarefoundation/autoware-documentation/pull/306/commits/cebfd3c8f86dd3b92ebc9f1607a148f972c35a2b :+1:

File: docs/installation/additional-settings-for-developers/index.md (hunk 2 above)
Comment (yukkysaito) [reply docs/installation/additional-settings-for-developers/index.md]: Fixed at https://github.com/autowarefoundation/autoware-documentation/pull/306/commits/cebfd3c8f86dd3b92ebc9f1607a148f972c35a2b :+1:

File: docs/installation/additional-settings-for-developers/index.md (hunk 3 above)
Comment (yukkysaito) [reply docs/installation/additional-settings-for-developers/index.md]: Fixed at https://github.com/autowarefoundation/autoware-documentation/pull/306/commits/cebfd3c8f86dd3b92ebc9f1607a148f972c35a2b :+1:

File: docs/installation/additional-settings-for-developers/index.md (hunk 4 above)
Comment (yukkysaito) [reply docs/installation/additional-settings-for-developers/index.md]: Fixed at https://github.com/autowarefoundation/autoware-documentation/pull/306/commits/cebfd3c8f86dd3b92ebc9f1607a148f972c35a2b :+1:

File: docs/installation/additional-settings-for-developers/index.md (hunk 5 above)
Comment (yukkysaito) [reply docs/installation/additional-settings-for-developers/index.md]: Fixed at https://github.com/autowarefoundation/autoware-documentation/pull/306/commits/cebfd3c8f86dd3b92ebc9f1607a148f972c35a2b :+1:

File: docs/installation/additional-settings-for-developers/index.md (hunk 6 above)
Comment (yukkysaito) [reply docs/installation/additional-settings-for-developers/index.md]: Fixed at https://github.com/autowarefoundation/autoware-documentation/pull/306/commits/cebfd3c8f86dd3b92ebc9f1607a148f972c35a2b :+1:

File: docs/support/troubleshooting.md (hunk 7 above)
Comment (yukkysaito) [reply docs/support/troubleshooting.md]: Fixed at https://github.com/autowarefoundation/autoware-documentation/pull/306/commits/cebfd3c8f86dd3b92ebc9f1607a148f972c35a2b :+1:

File: docs/installation/additional-settings-for-developers/index.md [hunk 17]
```diff
@@ -22,16 +22,46 @@ export RCUTILS_CONSOLE_OUTPUT_FORMAT="[{severity} {time}] [{name}]: {message} ({
...
+Please refer to [the official documentation](http://design.ros2.org/articles/ros_on_dds.html) for ROS2 networking concepts.
+
+### ROS 2 network setting
+
+ROS2 multicasts data on the local network by default. When developing in a company, etc., data flows over the local network and there is a possibility of collision.
+
+- Localhost-only communication
+- Same domain only communication on the local network
+
+Unless you plan to use multiple host computers on the local network, localhost-only communication is recommended.
```
[the remark this answers is not in the document]
Comment (yukkysaito) [reply docs/installation/additional-settings-for-developers/index.md:41]: How about this modification?
https://github.com/autowarefoundation/autoware-documentation/pull/306/commits/cebfd3c8f86dd3b92ebc9f1607a148f972c35a2b

[the remark this answers is not in the document]
Comment (kenji-miyake) [reply docs/installation/additional-settings-for-developers/index.md:41]: Looks good. Thank you!

File: docs/support/troubleshooting.md (hunk 8 above)
Comment (yukkysaito) [reply docs/support/troubleshooting.md:163]: okay :+1: 

Memo: AWSIM write this code in the tutorial.
https://tier4.github.io/AWSIM/GettingStarted/QuickStartDemo/

Comment (yukkysaito) [reply docs/support/troubleshooting.md:163]: @takam5f2 is it ok?

**Review by kenji-miyake — APPROVED** [review]

**Pull request merged**